base/cvd/cuttlefish/host/commands/run_cvd/launch/cvdalloc.cpp

```diff
@@ -67,10 +67,15 @@ bool Cvdalloc::Enabled() const { return instance_.use_cvdalloc(); }
 std::unordered_set<SetupFeature *> Cvdalloc::Dependencies() const { return {}; }
 
 Result<void> Cvdalloc::WaitForAvailability() {
-  LOG(INFO) << "cvdalloc (run_cvd): waiting to finish allocation.";
-  CF_EXPECT(cvdalloc::Wait(socket_, kCvdAllocateTimeout),
-            "cvdalloc (run_cvd): Wait failed");
-  LOG(INFO) << "cvdalloc (run_cvd): allocation is done.";
+  std::lock_guard<std::mutex> lock(availability_mutex_);
+  if (!available_) {
+    LOG(INFO) << "cvdalloc (run_cvd): waiting to finish allocation.";
+    CF_EXPECT(cvdalloc::Wait(socket_, kCvdAllocateTimeout),
```

Comment:
If two threads call this at roughly the same time the second one could wait for twice the `kCvdAllocateTimeout`: it will wait for the first thread to release the lock and then wait for the socket. If this happens one vm will fail in 30s while the other one will fail in 60s.

That can be avoided by making `available_` an enum with 3 possible values: `Pending`, `Done` or `Failed`. The first thread to wait for the socket changes it from `Pending` to the correct value and every subsequent call just reads that value.